### PR TITLE
Refactored Stats Iterator code to KryoLazyStatsUtils

### PR DESCRIPTION
**Pull request by spandanagrawal**
This pull requests refactors some generic Stats code from the StatsIterator class to a KryoLazyStatsUtils object

Signed-off-by: Spandan Agrawal <spandanagrawal@gmail.com>

### CODE REVIEW

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyDensityUtils.scala [hunk 1]
```diff
@@ -177,3 +183,64 @@ object KryoLazyDensityUtils {
     }
   }
 }
+
+object KryoLazyStatsUtils {
```
Comment (elahrvivaz) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyDensityUtils.scala]: let's put this in a separate file

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/process/stats/KryoLazyStatsIteratorProcessTest.scala [hunk 2]
```diff
@@ -14,15 +14,13 @@ import org.geotools.filter.text.ecql.ECQL
...
 import org.locationtech.geomesa.accumulo.TestWithDataStore
-import org.locationtech.geomesa.accumulo.iterators.KryoLazyStatsIterator._
 import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.index.utils.KryoLazyStatsUtils
 import org.locationtech.geomesa.utils.stats._
 import org.opengis.filter.Filter
 import org.specs2.mutable.Specification
 import org.specs2.runner.JUnitRunner
 
-import scala.collection.JavaConversions._
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/process/stats/KryoLazyStatsIteratorProcessTest.scala]: are you sure this import isn't needed? did you build this locally to verify?

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/stats/StatsIteratorProcess.scala [hunk 3]
```diff
@@ -73,8 +73,6 @@ class StatsIteratorProcess extends LazyLogging {
 
 class StatsVisitor(features: SimpleFeatureCollection, statString: String, encode: Boolean, properties: Array[String] = null)
     extends FeatureCalc with LazyLogging {
-
-  import scala.collection.JavaConversions._
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/stats/StatsIteratorProcess.scala]: ditto about this import

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyDensityUtils.scala (hunk 1 above)
Comment (spandanagrawal) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyDensityUtils.scala]: Done

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/process/stats/KryoLazyStatsIteratorProcessTest.scala (hunk 2 above)
Comment (spandanagrawal) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/process/stats/KryoLazyStatsIteratorProcessTest.scala]: Reverted this change

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/stats/StatsIteratorProcess.scala (hunk 3 above)
Comment (spandanagrawal) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/process/stats/StatsIteratorProcess.scala]: Reverted this as well

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyStatsUtils.scala [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import org.locationtech.geomesa.index.conf.QueryHints.{ENCODE_STATS, RichHints, STATS_STRING}
+import org.locationtech.geomesa.utils.collection.CloseableIterator
+import org.locationtech.geomesa.utils.geotools.GeometryUtils
+import org.locationtech.geomesa.utils.interop.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.stats.{Stat, StatSerializer}
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+/**
+  * Created by Spandan on 5/8/17.
+  */
```
Comment (jnh5y) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyStatsUtils.scala]: Can you remove this header and add the usual license header?  

You can test the headers locally with 'mvn license:check'.

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyDensityUtils.scala [hunk 5]
```diff
@@ -176,4 +176,4 @@ object KryoLazyDensityUtils {
       }
     }
   }
-}
+}
```
Comment (elahrvivaz) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyDensityUtils.scala]: can you revert this change to keep the newline at the end of the file? we try to do that for @tkunicki 's sake :)

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyStatsUtils.scala [hunk 6]
```diff
@@ -0,0 +1,75 @@
+package org.locationtech.geomesa.index.utils
```
Comment (elahrvivaz) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyStatsUtils.scala]: this file needs a license header

File: geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyStatsUtils.scala [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.index.conf.QueryHints.{ENCODE_STATS, RichHints, STATS_STRING}
+import org.locationtech.geomesa.utils.collection.CloseableIterator
+import org.locationtech.geomesa.utils.geotools.GeometryUtils
+import org.locationtech.geomesa.utils.interop.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.stats.{Stat, StatSerializer}
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+/**
+  * Created by Spandan on 5/8/17.
```
Comment (elahrvivaz) [inline geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyStatsUtils.scala]: should remove this (and ideally add some actual scaladocs)
